mailbox/api/src/main/java/org/apache/james/mailbox/model/search/PrefixedRegex.java

```diff
@@ -54,32 +59,12 @@ private boolean regexMatching(String name) {
 
     @Override
     public String getCombinedName() {
-        if (prefix != null && prefix.length() > 0) {
-            final int baseLength = prefix.length();
-            if (prefix.charAt(baseLength - 1) == pathDelimiter) {
-                if (regex != null && regex.length() > 0) {
-                    if (regex.charAt(0) == pathDelimiter) {
-                        return prefix + regex.substring(1);
-                    } else {
-                        return prefix + regex;
-                    }
-                } else {
-                    return prefix;
-                }
-            } else {
-                if (regex != null && regex.length() > 0) {
-                    if (regex.charAt(0) == pathDelimiter) {
-                        return prefix + regex;
-                    } else {
-                        return prefix + pathDelimiter + regex;
-                    }
-                } else {
-                    return prefix;
-                }
-            }
-        } else {
-            return regex;
-        }
+        String sanitizedPrefix = pathDelimiter.removeTrailingSeparatorAtTheEnd(prefix);
+        String sanitizedRegex = pathDelimiter.removeTrailingSeparatorAtTheBeginning(regex);
+        return pathDelimiter.join(
```

Comment:
the redundant separators handling could be  `join` responsibility, isn't it ?

Reply:
I would 100% but it brings a refactoring to a major behaviour change:

```
that.is..my.mailbox
```

Was valid before. And interpreted in a different way than:

```
that.is.my.mailbox
```

Should we consider both as the same?

Reply:
Do you mean that empty mailbox name is supported ?
from RFC 3501 : 
```
mailbox         = "INBOX" / astring
astring         = 1*ASTRING-CHAR / string
```
and string is non empty (nstring is used for empty)
So I think that the case of double delimiter is not valid